path/path_test.go

```diff
@@ -28,3 +28,27 @@ func TestPathParsing(t *testing.T) {
 		}
 	}
 }
+
+func TestPathCleaning(t *testing.T) {
+	cases := map[string]string{
+		// .. should strip the preceding path segment
```

Comment:
This is the same logic as [POSIX pathname resolution](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_12) and [relative URI resolution](https://tools.ietf.org/html/rfc3986#section-5.2.4).

User-specified objects named `.` and `..` would be un-addressable in most contexts already, and even if they were addressable in some contexts, it would be very confusing if they had a different meaning than in POSIX and in URIs.